openpype/hosts/maya/plugins/publish/extract_gltf.py

```diff
@@ -63,3 +67,80 @@ def process(self, instance):
         instance.data["representations"].append(representation)
 
         self.log.info("Extract GLB successful to: {0}".format(path))
+
+    def convert_gltf_shader(self, instance):
+        """
+        Converting StingrayPBS material to GLSL Shaders
+        specially for the glb export through Maya2GLTF plugin
+        """
+
+        meshes = cmds.ls(instance, type="mesh", long=True)
+        self.log.info("meshes: {}".format(meshes))
+        # load the glsl shader plugin
+        cmds.loadPlugin("glslShader", quiet=True)
+
+        for mesh in meshes:
+
+            # create glsl shader
+            glsl = cmds.createNode('GLSLShader')
+            glsl_shadingGrp = cmds.sets(name=glsl + "SG", empty=True,
+                                        renderable=True, noSurfaceShader=True)
+            cmds.connectAttr(glsl + ".outColor",
+                             glsl_shadingGrp + ".surfaceShader")
+
+            # load the maya2gltf shader
+            ogsfx_path = instance.context.data["project_settings"]["maya"]["publish"]["ExtractGLB"]["ogsfx_path"]  # noqa
+            if not os.path.exists(ogsfx_path):
+                self.log.info("Find the ogsfx shader file in "
+                              "default maya directory...")
+                # re-direct to search the ogsfx path in maya_dir
```

Comment:
This log should describe WHY it falls back to the `MAYA_APP_DIR` - what file wasn't found, what was missing?
It'd be even simpler if it wouldn't even _try_ to fallback but just explicitly only allow to use the path from settings.

Reply:
This is done like this mainly because there is no guarantee that setting up custom path from settings is right. Or somehow the user will just leave the path there with blank as they install the plugin in maya directory. 
The documentation of maya2glTF asks the user to copy and paste the files within MAYA_APP_DIR. If the user fails to put into the custom mode, the system tried to find if the ogsfx path located after the installation of maya2glTF,  and use that instead. I can use self.log.warning instead of self.log.info. Maybe I am thinking too much, but if I follow what you suggested in the comment in the similar PR(only allow to use the path from settings.) somehow the system can't find the path and error out. Maybe it's just my computer.

Reply:
I meant to say that IF the user had supplied a custom path in settings - the user would currently never be warned about that particular path not being found. It'd just see that it's searching the default maya directory.

E.g.

```python
            ogsfx_path = instance.context.data["project_settings"]["maya"]["publish"]["ExtractGLB"]["ogsfx_path"]  # noqa
            if not os.path.exists(ogsfx_path):
                if ogsfx_path:
                    # The settings that have a custom ogsfx_path set. Let's warn
                    # the user that it didn't find the path specified
                    self.log.warning("ogsfx shader file not found in {}".format(ogsfx_path))
                        
                self.log.info("Searching ogsfx shader file in "
                              "default maya directory...")
```